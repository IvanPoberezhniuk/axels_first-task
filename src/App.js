import './App.scss';
import Header from './components/particials/Header';
import Footer from './components/particials/Footer';
import styled from 'styled-components';
import GalleryPage from './pages/GalleryPage';


const StyledDiv = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
`;

function App() {
  return (
    <StyledDiv className="App">
      <Header/>
      <main>
        <GalleryPage/>
      </main>
      <Footer/>
    </StyledDiv>
  );
}

export default App;
