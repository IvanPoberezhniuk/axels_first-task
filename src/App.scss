$theme-colors: ("primary": #4997d0);
$enable-rounded: false;

@import "~bootstrap/scss/bootstrap";

.App {
  text-align: center;
}
