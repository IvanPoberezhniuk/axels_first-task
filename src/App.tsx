import { Switch, useLocation, Route } from 'react-router-dom';

import { Footer, Header, ImageModal } from './components';

import { routes, RouteWithSubRoutes } from './routes/routes';
import { AppDiv } from './styled/App';
import './App.scss';

const App = () => {
  const location = useLocation();
  // @ts-ignore
  const backgroundPage = location.state && location.state.background;

  return (
    <AppDiv>
      <Header />
      <main>
        <Switch location={backgroundPage || location}>
          {routes.map((route, i) => (
            <RouteWithSubRoutes key={i} {...route} />
          ))}
        </Switch>
        {backgroundPage && (
          <Route path='/gallery/:id' children={<ImageModal />} />
        )}
      </main>
      <Footer />
    </AppDiv>
  );
};

export default App;
