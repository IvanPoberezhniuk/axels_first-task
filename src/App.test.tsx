import App from './App';

import { BrowserRouter } from 'react-router-dom';
import { render } from 'enzyme';

let component: cheerio.Cheerio;
const setUp = () => render(<BrowserRouter><App /></BrowserRouter>);

beforeEach(() => {
  component = setUp();
});

describe('APP COMPONENT', () => {
  it('SNAPSHOT', () => {
    expect(component).toMatchSnapshot();
  });
});

