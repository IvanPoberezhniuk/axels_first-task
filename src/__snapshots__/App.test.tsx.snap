// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP COMPONENT SNAPSHOT 1`] = `
<div
  class="sc-jrAGrp gzEPbS"
>
  <header
    class="sc-gsTCUz jmMyoC"
  >
    <h1>
      TEST APP
    </h1>
  </header>
  <main />
  <footer
    class="sc-bdfBwQ jygzMn"
  >
    <small>
      Â© 2018-2019
    </small>
  </footer>
</div>
`;
